### PR TITLE
Edit Logrotate: Reduce Complexity & hard set 100M size limit

### PR DESCRIPTION
**Pull request by rootzoll**
adresses: #4023

### CODE REVIEW

**Comment by rootzoll** [comment]
The goal of this PR is to rotate all logs on a size limit & flush old rotates on reboot - to prevent logs filling up the sd card finally.

**Comment by openoms** [comment]
rebased on the updated image builds

File: build_sdcard.sh [hunk 1]
```diff
@@ -491,75 +491,45 @@ fi
...
 sed -i "s/^#SystemMaxFileSize=.*/SystemMaxFileSize=50M/g" /etc/systemd/journald.conf
 
-# change log rotates
-# see https://github.com/rootzoll/raspiblitz/issues/394#issuecomment-471535483
+## LOG ROTATION
+
+# GLOBAL for all logs: /etc/logrotate.conf
+echo "# Optimizing log files: rotate daily max 100M, keep 4 days & compress old" >> ${logFile}
+sed -i "s/^weekly/daily size 100M/g" /etc/logrotate.conf >> ${logFile} 2>&1
+sed -i "s/^#compress/compress/g" /etc/logrotate.conf >> ${logFile} 2>&1
```
Comment (openoms) [inline build_sdcard.sh]: The `${logFile}` is not used here

```suggestion
echo "# Optimizing log files: rotate daily max 100M, keep 4 days & compress old"
sed -i "s/^weekly/daily size 100M/g" /etc/logrotate.conf
sed -i "s/^#compress/compress/g" /etc/logrotate.conf
```

File: home.admin/_bootstrap.sh [hunk 2]
```diff
@@ -167,6 +167,9 @@ sleep 5
 # CLEANING BOOT SYSTEM
 ################################
 
+# force a rotation of every log on boot
+logrotate -f
```
Comment (openoms) [inline home.admin/_bootstrap.sh]: needs a configfile like:
```
logrotate -f /etc/logrotate.d/rsyslog
```
but getting an error on a running system:
```
root@raspberrypi:~# logrotate -f /etc/logrotate.d/rsyslog
Failed to reload rsyslog.service: Job type reload is not applicable for unit rsyslog.service.
error: error running shared postrotate script for '/var/log/syslog
/var/log/mail.info
/var/log/mail.warn
/var/log/mail.err
/var/log/mail.log
/var/log/daemon.log
/var/log/kern.log
/var/log/auth.log
/var/log/user.log
/var/log/lpr.log
/var/log/cron.log
/var/log/debug
/var/log/messages
'
```

File: build_sdcard.sh [hunk 3]
```diff
@@ -491,75 +491,45 @@ fi
...
-  weekly
+  size 100M
   missingok
-  notifempty
   compress
   delaycompress
   sharedscripts
   postrotate
-    invoke-rc.d rsyslog rotate > /dev/null
+    systemctl reload rsyslog
```
Comment (openoms) [inline build_sdcard.sh]: this doesn't work:
```
sudo systemctl reload rsyslog
Failed to reload rsyslog.service: Job type reload is not applicable for unit rsyslog.service.
```

Comment (openoms) [reply build_sdcard.sh]:
```suggestion
    service logrotate restart
```

File: home.admin/_bootstrap.sh (hunk 2 above)
Comment (openoms) [reply home.admin/_bootstrap.sh]: working when using service logrotate restart in the rsyslog file. Resolving this.

Comment (openoms) [reply home.admin/_bootstrap.sh]:
```suggestion
logrotate -f /etc/logrotate.d/rsyslog
```

**Comment by openoms** [comment]
Tested, but needs some changes.

File: home.admin/_bootstrap.sh (hunk 2 above)
Comment (rootzoll) [reply home.admin/_bootstrap.sh:171]: The idea is here that it should do it for all logs - not just the one in /etc/logrotate.d/rsyslog

**Pull request merged**